File: vm/tests/all/interop/inheritMethods/inheritMethods.cpp
//inheritMethodsClass.cpp
#include<polyglot.h>

class ConstAdder {
	protected:
		int constObj;
	public:
	ConstAdder(int x);
	virtual int process(int y);
	virtual int getConstObj();
};

DECLARE_POLYGLOT_CLASS(ConstAdder);

class ConstMultiplier extends ConstAdder {
	public:
		ConstMultiplier(int x);
		int process(int y);
};

DECLARE_POLYGLOT_CLASS(ConstMultiplier);

ConstAdder::ConstAdder(int x) {
	constObj = x;
}

int ConstAdder::process(int y) {
	return y+constObj;
}

int ConstAdder::getConstObj() {
	return constObj;
}

ConstMultiplier::ConstMultiplier(int x):ConstAdder(x) {}

int ConstMultiplier::process(int y) {
	return y*constObj;
}

File: vm/tests/all/interop/inheritMethods/inheritMethods.py
#inheritMethodsClass.py

class ConstAdder:

	def __init__(self, x):
		self.constObj = x
	
	def process(self, y):
		return y+self.constObj

	def getConstObj(self):
		return self.constObj


class ConstMultiplier(ConstAdder):

	def __init__(self, x):
		ConstAdder.__init__(self, x)
	
	def process(self, y):
		return y*self.constObj
